packages/cli/src/tools/config/readConfigFromDisk.js

```diff
@@ -82,7 +98,9 @@ export function readLegacyDependencyConfigFromDisk(
       hooks: config.commands,
       params: config.params,
     },
-    commands: [].concat(config.plugin || []),
+    commands: []
+      .concat(config.plugin || [])
+      .map(p => loadProjectCommand(rootFolder, p)),
```

Comment:
The RNPM config is confusing. Can be `plugin: ['a.js']`, `plugin: 'a.js'` and contents of `a.js` can be either `module.exports = {}` or `module.exports = [{}]`. This code tries to handle that.

Reply:
Hm, so the type is wrong, no? If I understand you correctly `p: Array<string> | string`, while `loadProjectCommand` assumes only `Array<string>`. I also don't see any handling for `module.exports = [{}]` scenario, am I missing something?

Reply:
Yup. We use the `concat` on the line above to convert it to an array (concat takes an array or an item to append). Then, we do the same with concat(require) inside `loadProjectCommand`